### DISCUSSION TITLE
Get all documents in collection one time

### DISCUSSION THREAD

**Discussion by stephane2018** [appwrite/appwrite#4807] [Q&A] [answered]
I have another question, I wanted to know if it is possible to get all the documents of a collection. I have searched in the doc for a way to do this but I can't find it. Apparently you can only retrieve data by applying a limit and the default is 25 I think. But I want to retrieve the whole collection at once.

_Originally posted by @stephane2018 in https://github.com/appwrite/appwrite/discussions/4789#discussioncomment-4241679_

**Comment by netapy** [comment, marked as answer]
It is not possible yet.
The only way is to loop through the database with the query `Query.limit(100)` and stop when the query result length == 0. 
Edit : Please refer to our [Pagination](https://appwrite.io/docs/pagination) docs for more info on this. (thanks Steven)

**Reply by stnguyen90** [reply]
Please refer to our [Pagination](https://appwrite.io/docs/pagination) docs for more info on this.

**Comment by stephane2018** [comment]
ok thanks

**Comment by Dosbodoke** [comment]
Currently I have an list of flag ids (collection `flag`) on my collection `task` and I need to generate a report for this task, for that, I need to get, for each flag, all associated `note`. The problem is that I can't just pass `sdk.Query.equal("flagId", flagIdList)` to my Query because it limits the number of items so I need to chunk my ids and retrieve in many calls.

I wished this limit could be disabled through some configuration (maybe only for functions)

```
  const data = JSON.parse(APPWRITE_FUNCTION_DATA)
  const task = await database.getDocument(
    APPWRITE_DATABASE_ID,
    "task",
    data["taskId"]
  )

  const notes = {}
  const profileData = {}
  const chunkSize = 100

  // Itarate over the flagIdlist and get the notes of the current chunk.
  // For each note, check if the writer data is stored, if not get and store it
  // to reduce the number of requests.
  for (let i = 0; i < task.flagIdList.length; i += chunkSize) {
    const flagIdChunk = task.flagIdList.slice(i, i + chunkSize)
    const notesChunk = await database.listDocuments(
      req.variables["APPWRITE_DATABASE_ID"],
      "note",
      [sdk.Query.equal("flagId", flagIdChunk), sdk.Query.limit(chunkSize)]
    )
    notesChunk.documents.forEach(async (document) => {
      const writerId = document["writerId"]

      if (!profileData[writerId]) {
        const writer = await users.get(writerId)
        profileData[writerId] = writer.name
      }

      notes[document.flagId] = notes[document.flagId]
        ? [
            ...notes[document.flagId],
            { ...document, writerName: profileData[document.writerId] },
          ]
        : [{ ...document, writerName: profileData[document.writerId] }]
    })
  }

  res.json(
    {
      notes,
    },
  )
  ```

**Reply by rafagazani** [reply]
I had the same problem as you. Generate reports from appwrite data. Using the appwrite api for this I found unfeasible.
My solution was to create a function to send all documents from appwrite to postgres. This function is triggered by the created, update and delete events.

I connected the metabase to Postgres to create the reports.
I recommend uploading another vps to keep Postgres and metabase
It was the best solution I found.

I don't recommend connecting directly to appwrite's mysql, even though I optimized my queries I managed to crash the server.
I say in millions of records this was the best solution so far
